src/client.rs

```diff
@@ -17,7 +17,11 @@ impl<T: DaprInterface> Client<T> {
     ///
     /// * `addr` - Address of gRPC server to connect to.
     pub async fn connect(addr: String) -> Result<Self, Error> {
-        Ok(Client(T::connect(addr).await?))
+        // Get the Dapr port to create a connection
+        let port: u16 = std::env::var("DAPR_GRPC_PORT")?.parse()?;
```

Comment:
It's better to use `match` and handle the errors here explicitly (we can revert error.rs then). In case of error parsing the environment variable, we can log and fallback to a standard default.